tests/lua/cartesi-machine-tests.lua

```diff
@@ -730,33 +740,47 @@ local function run_parallel(contexts)
                 ctx.cycles = ctx.target.run(machine, ctx, 2 * ctx.expected_cycles)
                 check_test_result(ctx)
                 ctx.target.destroy(machine)
-                unistd._exit(0)
+                if ctx.failed then
+                    unistd._exit(3)
```

Comment:
Oh so it was never exiting with an error when a test failed, at the time I wrote this I remember that it was working, I guess some recent change broke it.

Reply:
Removed, will be correctly fixed on another PR